### PR TITLE
Future clippy

### PR DESCRIPTION
**Pull request by eranrund**
### Motivation

In preparation for a Rust version upgrade I ran into a bunch of small things future clippy will complain about, so here are fixes for them.

### In this PR
* `impl Into` -> `impl From`
* `ORM` -> `Orm`
* Getting rid of some redundant usages of the question-mark operator
* Getting rid of some unnecessary slices
* Initializing structs in a way that pleases clippy
* Removing an unnecessary `transmute` call
* Remove unnecessary `#[no_mangle]` attributes

### CODE REVIEW

File: fog/ocall_oram_storage/trusted/src/lib.rs [hunk 1]
```diff
@@ -550,11 +550,8 @@ mod helpers {
 
 // This stuff must match edl file
 extern "C" {
-    #[no_mangle]
     fn allocate_oram_storage(count: u64, data_size: u64, meta_size: u64, id: *mut u64);
```
Comment (garbageslam) [inline fog/ocall_oram_storage/trusted/src/lib.rs:553]: Is it actually okay to remove no mangle here? I thought that is required for linkage to work. Is it implied by extern C or something?

File: fog/ocall_oram_storage/untrusted/src/lib.rs [hunk 2]
```diff
@@ -170,7 +170,7 @@ pub unsafe extern "C" fn allocate_oram_storage(
         data_size as usize,
         meta_size as usize,
     ));
-    let id: u64 = core::mem::transmute(Box::into_raw(result));
+    let id = Box::into_raw(result) as u64;
```
Comment (garbageslam) [inline fog/ocall_oram_storage/untrusted/src/lib.rs:173]: Thank you!

File: fog/ocall_oram_storage/trusted/src/lib.rs (hunk 1 above)
Comment (eranrund) [reply fog/ocall_oram_storage/trusted/src/lib.rs:553]: Yup, I believe it is not needed: https://doc.rust-lang.org/std/keyword.extern.html

File: fog/ocall_oram_storage/untrusted/src/lib.rs (hunk 2 above)
Comment (eranrund) [reply fog/ocall_oram_storage/untrusted/src/lib.rs:173]: Thank clippy 😛

**Review by garbageslam — APPROVED** [review]

**Review by mfaulk — APPROVED** [review]

**Pull request merged**